test/test_torchrun.py

```diff
@@ -0,0 +1,91 @@
+# Owner(s): ["module: unknown"]
+
+from torch.testing._internal.common_utils import TestCase, run_tests
+import os
+import shutil
+import subprocess
+
+class TestTorchrun(TestCase):
+    def test_jemalloc(self):
+        if os.path.isdir('torchrun_logs'):
+            shutil.rmtree('torchrun_logs')
+        num = 0
+        loc = os.path.dirname(os.path.abspath(__file__))
+        with subprocess.Popen("torchrun --inference --nprocs 1 --enable_jemalloc --log_path torchrun_logs --disable_numactl \
```

Comment:
To keep the test assertions close to the logic that it is testing, I recommend testing `torch.utils.inference.run` as a module rather than through `torchrun --inference` CLI. 

For `torch.utils.run` you should definitely add a test that asserts that passing `--inference` indeed calls `torch.utils.inference.run` and passing nothing calls `torch.distributed.run`

Reply:
done